### PR TITLE
🔀🔖 4.7.7

### PR DESCRIPTION
**Pull request by jxn-30**
**Releasenotes**

* 🐛 [extendedCallList/eventMissions] fix search when mission is updated
* 🌐 updated POI-Lists to be up-to-date in all versions
* 📝 [docs] updated docs on how to install LSSM on Android devices *(mobile devices are still not officially supported but LSSM mostly works fine)*
* 🚸 [extendedBuilding/schoolsBuildingFilter] improve position of filter to avoid jumps
* ✨🇩🇪 support new Großwache, Großkrankenhaus and Großgewahrsam extensions
* ✨🇳🇱 support Waterwerper vehicle and extension
* ✨ [extendedCallList/sortMissions] sort by ZIP-Code
* ✨ [extendedMap/buildingComplexes] add button to open a free school
* ✨ [hotkeys] add `schoolings.educate` hotkey

---

**Releasenotes for translations**
```md
* 🐛 [$s('extendedCallList.eventMissions')] fix search when mission is updated
* 🌐 updated POI-Lists to be up-to-date in all versions
* 📝 [wiki] updated docs on how to install LSSM on Android devices: [Using LSSM on a smartphone](https://docs.lss-manager.de/en_US/faq.html#using-lssm-on-a-smartphone)  *(mobile devices are still not officially supported but LSSM mostly works fine)*
* 🚸 [$s('extendedBuilding.schoolsBuildingFilter')] improve position of filter to avoid jumps
* ✨🇩🇪 support new Großwache, Großkrankenhaus and Großgewahrsam extensions
* ✨🇳🇱 support Waterwerper vehicle and extension
* ✨ [$s('extendedCallList.sortMissions')] sort by ZIP-Code
* ✨ [$s('extendedMap.buildingComplexes')] add button to open a free school
* ✨ [$m('hotkeys')] add `schoolings.educate` hotkey
```

### CODE REVIEW

File: src/modules/dashboard/components/building-types.vue [hunk 1]
```diff
@@ -134,7 +134,7 @@ export default Vue.extend<
                     color,
                     rows: Object.fromEntries(
                         Object.values(buildings).flatMap(buildingType => {
-                            const removeNull = <S>(
+                            const removeNull = <S,>(
```
Comment (Cr4zyc4k3) [inline src/modules/dashboard/components/building-types.vue:137]: Typo?

File: src/modules/extendedCallWindow/assets/arrHover.ts [hunk 2]
```diff
@@ -361,7 +361,7 @@ export default (
             ...(
                 JSON.parse(group.getAttribute('vehicles') ?? '[]') as [
                     number,
-                    string
+                    string,
```
Comment (Cr4zyc4k3) [inline src/modules/extendedCallWindow/assets/arrHover.ts:364]: Typo?

File: src/modules/hotkeys/assets/commands/schoolings.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import type { Empty, Scope } from 'typings/modules/Hotkeys';
+
+export default <Scope<Empty, [], [], true>>{
+    educate() {
+        // #multiple_commits is the button added by https://forum.leitstellenspiel.de/index.php?thread/23203-mehr-als-4-klassenr%C3%A4ume-zeitgleich-nutzen/
+        document
+            .querySelector<HTMLInputElement | HTMLSpanElement>(
```
Comment (Cr4zyc4k3) [inline src/modules/hotkeys/assets/commands/schoolings.ts:7]: Why do we check on  a specific script?

File: src/modules/shareAlliancePost/assets/util.ts [hunk 4]
```diff
@@ -13,20 +13,28 @@ export function getCityFromAddress(address: string): string {
...
-            // matches all Zip-styles of the countries supported by LSSM
-            /^((\d{4} ?[A-Z]{2})|((\d{4}|\d{2})[ -]\d{3})|(\d{3} \d{2})|\d+|([\dA-Z]{2,4} [\dA-Z]{3}))/u,
-            ''
-        )
-        .trim();
+    return city.replace(ZIP_REGEX, '').trim();
+}
+
+/**
+ * A method to extract the ZIP Code from a city name.
```
Comment (Cr4zyc4k3) [inline src/modules/shareAlliancePost/assets/util.ts:31]: JSdoc, @jxn-30 are you ill? :wink:

Comment (jxn-30) [reply src/modules/shareAlliancePost/assets/util.ts:31]: yes ;)

File: src/modules/hotkeys/assets/commands/schoolings.ts (hunk 3 above)
Comment (jxn-30) [reply src/modules/hotkeys/assets/commands/schoolings.ts:7]: for compatibility reason. The script removes the default button thus the hotkey wouldn't work anymore.

**Review by kdev — APPROVED** [review]

**Pull request merged**